src/posts/lucky_0.25.cr

```diff
@@ -0,0 +1,370 @@
+    This method was called `with_defaults` before, but has been renamed for some clarity on what it does.
+
+    ### Changes to the Query objects
+
+    The `Query` objects also got a few new changes. The main one being that queries no longer mutate the object. Prior to this release, appending any
+    query method would mutate the original object. This made building queries really nice and easy, but presented a problem that many faced. For example:
+
+    ```crystal
+    q = UserQuery.new
+    if current_user.admin?
+      q.admin(true)
+    end
+
+    # this would fail because we've already mutated the query
+    # on the count, and can no longer query with the username ordering
+    user_total_count = q.select_count
+    users = q.username.asc_order
+    ```
+
+    In that case, you had to make sure you cloned the query, and they were in the correct order. Now we no longer mutate the query, but this requires changing your
+    code just slightly.
+
+    ```crystal
+    q = UserQuery.new
+    if current_user.admin?
+      q = q.admin(true)
+    end
+
+    user_total_count = q.select_count
+    users = q.username.asc_order
+    ```
+
+    With this change, this also means that defining default queries in your query class `initialize` need to change. To set a default, you'll now use the `defaults` method.
+
+    ```crystal
+    class UserQuery < User::BaseQuery
+      def initialize
+        defaults &.admin(true)
+      end
+    end
+    ```
+
+    ### Type-safe WHERE "OR"
+
+    This was one of the most requested methods we got! We held off since adding this in makes SQL queries quite a bit more complex, but we finally got it! (**sort of... see note below)
+
+    ```crystal
+    # WHERE users.name = 'Billy' OR users.name = 'Kelly'
+    UserQuery.new.name("Billy").or(&.name("Kelly"))
+    ```
+
+    As you can see, you call the `or` method which passes in the instance of the query object allowing you to chain additional WHERE clauses.
+
+    **NOTE: The above example works great, and will cover plenty of the queries that you may have, but we currently make no assumptions on where to place parenthesis for scoping
+    order of operations. Take this for example:
+
+    ```crystal
+    UserQuery.new.name("Billy").or(&.name("Kelly").age.gte(51)).or(&.admin(true))
+    ```
+
+    This query would generate `WHERE name = 'Billy' OR name = 'Kelly' AND age >= 51 OR admin = true`, and that may not give you the result you expect. We are working on a solution,
+    but for now if you need more complex control, you can pass raw SQL to `UserQuery.new.where("name = ? OR (name = ? && age >= ?)", "Billy", "Kelly", 51)`.
+
+    ### Model Association updates
+
+    When you're using a RDBMS like postgres, table associations become very important for properly structuring your data. There were several bugs fixed when it comes to
+    associations giving us a lot more power and control over our data.
```

Comment:
```suggestion
    associations giving you a lot more power and control over your data.
```

Reply:
I was trying (slightly) to avoid `us` or `you` just making it person-less. But I do like `you` over `us`.